lib/internal/vm.js

```diff
@@ -97,22 +101,65 @@ function internalCompileFunction(code, params, options) {
   }
 
   if (importModuleDynamically !== undefined) {
-    validateFunction(importModuleDynamically,
-                     'options.importModuleDynamically');
-    const { importModuleDynamicallyWrap } = require('internal/vm/module');
-    const wrapped = importModuleDynamicallyWrap(importModuleDynamically);
-    const func = result.function;
-    const { registerModule } = require('internal/modules/esm/utils');
-    registerModule(func, {
-      __proto__: null,
-      importModuleDynamically: wrapped,
-    });
+    registerImportModuleDynamically(result.function, importModuleDynamically);
   }
 
   return result;
 }
 
+function makeContextifyScript(code,
+                              filename,
+                              lineOffset,
+                              columnOffset,
+                              cachedData,
+                              produceCachedData,
+                              parsingContext,
+                              hostDefinedOptionId,
+                              importModuleDynamically) {
+  let script;
+  // Calling `ReThrow()` on a native TryCatch does not generate a new
+  // abort-on-uncaught-exception check. A dummy try/catch in JS land
+  // protects against that.
+  try { // eslint-disable-line no-useless-catch
+    script = new ContextifyScript(code,
```

Comment:
should the try/catch be done inside `ContextifyScript`, then, so that callsites don't need to repeat it?

Reply:
ContextifyScript is implemented in C++ which ReThrow after error decoration and that’s why we need to rethrow again in JS. I tried using an InternalScript class before but that would result in changes to the stack frames so I would rather just be dumb and repeat it here

Reply:
Can ReThrow be made to generate an uncaught exception check? This just feels like a huge footgun that any errors can be thrown and uncaught that wouldn’t hit the handler.